libc/sem.cc

```diff
@@ -76,3 +105,81 @@ int sem_trywait(sem_t* s)
         return libc_error(EAGAIN);
     return 0;
 }
+
+//TODO: when are these initialized?
+static std::unordered_map<std::string, indirect_semaphore> named_semaphores;
+static mutex named_semaphores_mutex;
+
+OSV_LIBC_API
+sem_t *sem_open(const char *name, int oflag){
+    SCOPE_LOCK(named_semaphores_mutex);
+    std::unordered_map<std::string, indirect_semaphore>::iterator iter = 
+                                named_semaphores.find(std::string(name));
+    if(iter != named_semaphores.end()){
+        iter->second.add_reference();
+        return reinterpret_cast<sem_t*>(&(iter->second));
+    }
+    
+    errno = ENOENT;
+    return SEM_FAILED;
+}
+
+OSV_LIBC_API
+sem_t *sem_open(const char *name, int oflag, mode_t mode, unsigned int value){
+    SCOPE_LOCK(named_semaphores_mutex);
+    std::unordered_map<std::string, indirect_semaphore>::iterator iter = 
+                                named_semaphores.find(std::string(name));
+    
+    if(oflag & O_CREAT){
+        if(iter != named_semaphores.end()){
+            if(oflag & O_EXCL){
+                errno = EEXIST;
+                return SEM_FAILED;
+            }
+
+            iter->second.add_reference();
+            return reinterpret_cast<sem_t*>(&(iter->second));
+        }
+        
+        if(value > SEM_VALUE_MAX){
+            errno = EINVAL;
+            return SEM_FAILED;
+        }
+        /*
+         * Ignoring mode bits bits for now. Do we actually want to worry about 
```

Comment:
I do not think we need to worry about it.

Reply:
Would you like me to remove this comment, or leave/modify it as a design choice?

Reply:
Please remove this comment.